torch/testing/_internal/common_methods_invocations.py

```diff
@@ -188,6 +188,8 @@ def _np(t):
                 return tuple(map(to_numpy, x))
             elif isinstance(x, dict):
                 return {k: to_numpy(v) for k, v in x.items()}
+            elif isinstance(x, torch.dtype):
+                return torch.empty(0, dtype=x).numpy().dtype
```

Comment:
Interesting move :D
Maybe we can re-use: https://github.com/pytorch/pytorch/blob/421d8f86b6def536df18371a5da2f5df4de6e262/torch/testing/_internal/common_utils.py#L455 ?

Reply:
> I'm not super happy with all the tolerance overrides but I think the PR is good :)

Could you elaborate more on your concerns about the tolerance overrides? Do you think we should skip the tests instead? Or if it a concern with the numerical instability they expose?

Reply:
If so many functions need to override the default value, maybe the default is not good?
Or we want to run these in double to make sure they are correct (like we do for gradcheck)?

Reply:
The problem is mostly with sum and prod for float16. I changed the decorator to override the tolerance for float16 for all tests. Some tests are specifically testing numerical stability so it's expected we need to override the tolerance, this is also a way to document numerical instability.

Reply:
I changed the large reference tests from float32 to float64